tesseract-core/src/query.rs

```diff
@@ -425,30 +425,96 @@ impl FromStr for GrowthQuery {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "and" => Operator::And,
+            "or" => Operator::Or,
+            _ => bail!("Could not parse sort direction"),
+        })
+    }
+}
+
+fn get_filter(filter_split: Vec<String>, split_int: usize) -> Result<FilterQuery, Error> {
+    let by_mea_or_calc = filter_split[0].parse::<MeaOrCalc>()?;
+    let constraint = join(&filter_split[1..split_int], ".").parse::<Constraint>()?;
+    let operator = Some(filter_split[split_int].parse::<Operator>()?);
+    let constraint2 = Some(join(&filter_split[split_int+1..], ".").parse::<Constraint>()?);
+    Ok(FilterQuery {
+        by_mea_or_calc,
+        constraint,
+        operator,
+        constraint2,
+    })
+}
+
 /// For filtering on a measure after Top is calculated (wrapper around end aggregation)
 #[derive(Debug, Clone)]
 pub struct FilterQuery {
     pub by_mea_or_calc: MeaOrCalc,
     pub constraint: Constraint,
+    pub operator: Option<Operator>,
+    pub constraint2: Option<Constraint>
 }
 
 // Currently only allows one sort_measure
 impl FromStr for FilterQuery {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match &s.split(",").collect::<Vec<_>>()[..] {
-            [by_mea, constraint] => {
-
-                let by_mea_or_calc = by_mea.parse::<MeaOrCalc>()?;
-                let constraint = constraint.parse::<Constraint>()?;
-
-                Ok(FilterQuery {
-                    by_mea_or_calc,
-                    constraint,
-                })
-            },
-            _ => bail!("Could not parse a filter query"),
+        if s.contains(".and.") || s.contains(".or.") {
+            let filter_split: Vec<String> = s.split(".").map(|f| f.to_string()).collect();
+            match filter_split.len() {
+                6 => {
+                    get_filter(filter_split, 3)
+                },
+                7 => {
+                    if filter_split[3] == "or" || filter_split[3] == "and" {
+                        get_filter(filter_split, 3)
+                    } else {
+                        get_filter(filter_split, 4)
+                    }
+                },
+                8 => {
+                    get_filter(filter_split, 4)
+                },
+                _ => bail!("Could not parse a filter query"),
+            }
+        } else {
+            match &s.splitn(2, ".").collect::<Vec<_>>()[..] {
+                [by_mea, constraint] => {
+
+                    let by_mea_or_calc = by_mea.parse::<MeaOrCalc>()?;
+                    let constraint = constraint.parse::<Constraint>()?;
+
+                    Ok(FilterQuery {
+                        by_mea_or_calc,
+                        constraint,
+                        operator: None,
+                        constraint2: None
+                    })
+                },
+                _ => bail!("Could not parse a filter query"),
+            }
```

Comment:
this might be simplified with a regular expression?

Reply:
It can be solved using the regex or also without regex and in a simpler format than what I have done, I would love to contribute if that is okay.

Reply:
@jspeis just made a commit solving this issue just make sure to test it before merging it the logic works it should not have any issues. I was not able to run the test because of a lack of Openssl in the windows system and also I don't have a local clickhouse set up in my current system as of right now. Let me know if any other changes are needed

Reply:
Wow, great! Thanks for going above & beyond Jay! I'll take a look